clippy_lints/src/strings.rs

```diff
@@ -152,7 +161,21 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for StringLitAsBytes {
             if path.ident.name == "as_bytes" {
                 if let ExprKind::Lit(ref lit) = args[0].node {
                     if let LitKind::Str(ref lit_content, _) = lit.node {
-                        if lit_content.as_str().chars().all(|c| c.is_ascii()) && !in_macro(args[0].span) {
+                        let callsite = snippet(cx, args[0].span.source_callsite(), "");
+                        let expanded = format!("\"{}\"", lit_content.as_str());
```

Comment:
I wasn't sure the best way to handle this, the content removes the quotation marks, I'm worried that this will break for raw strings when it shouldn't

Reply:
Could you add a test for raw strings, then we'll see how to handle them.

Reply:
It doesn't lint the raw strings at all as currently written because the poor reproduction of the expanded lit doesn't satisfy the first expression in the else if `callsite == expanded`.

I'm gonna try to inspect the members of ExprKind and LitKind to see if I can find some place that stores the terminators or w/e they're called for the raw string. Otherwise I can also just search the callsite for an `r(.*)"` type pattern and reuse that when constructing the expanded literal.

Reply:
`LitKind::Str` has as second parameter the `StrStyle`. If it is a `StrStyle::Raw(u16)`, the `u16` will tell you how many `#` were used. I hope this helps. [Docs](https://doc.rust-lang.org/nightly/nightly-rustc/syntax/ast/enum.StrStyle.html#variant.Raw)

Reply:
Variants `Cooked` and `Raw`, beast. This is perfect ty for the heads up.